cores/esp32/IPAddress.h

```diff
@@ -54,26 +52,27 @@ class IPAddress: public Printable
     bool fromString(const char *address);
     bool fromString(const String &address) { return fromString(address.c_str()); }
 
+
     // Overloaded cast operator to allow IPAddress objects to be used where a pointer
     // to a four-byte uint8_t array is expected
     operator uint32_t() const
     {
-        return _address.dword;
+        return reinterpret_cast<const uint32_t&>(_address.front());
```

Comment:
Idem, reinterpret_cast should be used with utmost care, especially with alignment.